src/Microsoft.DocAsCode.MarkdigEngine.Extensions/LineNumber/LineNumberExtension.cs

```diff
@@ -3,23 +3,34 @@
 
 namespace Microsoft.DocAsCode.MarkdigEngine.Extensions
 {
-    using Markdig.Parsers;
+    using Markdig;
+    using Markdig.Renderers;
     using Markdig.Renderers.Html;
     using Markdig.Syntax;
     using Markdig.Syntax.Inlines;
 
-    public class LineNumberExtension
+    public class LineNumberExtension : IMarkdownExtension
     {
         public const string EnableSourceInfo = "EnableSourceInfo";
 
-        public static ProcessDocumentDelegate GetProcessDocumentDelegate(string filePath)
+        private readonly MarkdownContext _context;
+
+        public LineNumberExtension(MarkdownContext context) => _context = context;
+
+        public void Setup(MarkdownPipelineBuilder pipeline)
         {
-            return (MarkdownDocument document) =>
+            pipeline.PreciseSourceLocation = true;
+            pipeline.DocumentProcessed += document =>
             {
-                AddSourceInfoInDataEntry(document, filePath);
+                AddSourceInfoInDataEntry(document, _context.GetFilePath(_context.File));
             };
         }
 
+        public void Setup(MarkdownPipeline pipeline, IMarkdownRenderer renderer)
+        {
+
+        }
```

Comment:
what's that for?

Reply:
in v2, filePath is a `RelativePath`, `RelativePath.ToString()` starts with '~/' if the path is from working directory. The expected source info should not contain '~/'